core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -319,7 +319,7 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
 
       case e: Exception if env.isStopped =>
         logDebug("Exception thrown after context is stopped", e)
-        null.asInstanceOf[OUT]  // exit silently
+        throw new SparkException("Spark session has been stopped", e)
```

Comment:
is this behavior consistent with java tasks?

Reply:
I checked the code there are two places there `env.isStopped` is used:
* PythonRunner will complete silently when env.isStopped = true
* When retry failed tasked and env.isStopped, Spark will ignore RejectedExecutionException.

As far as java tasks go, during shutdown I don't think there are code that make them exit silently.

Reply:
@cloud-fan Thoughts?

Reply:
It seems like java task doesn't check `env.isStopped` when task failed, and only check it when retry failed.

Does python task have retry mechanism?

Reply:
Python task has the same retry mechanism as Java - when a python task fails, the corresponding java task (EvalPythonExec/FlatMapGroupsInPandasExec) also fails and the java task is retried.

I think the python/java behavior would be consistent with this patch - when spark session is stopped, java task will receive an when reading from python task output, and then throw an SparkException. The java task will then be enqueued for retry as which point it will be the same as other java tasks that are enqueued for retry during shutdown.

Reply:
then shall we just remove this case to be totally consistent with java task? i.e. no logging here and no extra SparkException to wrap the original exception.

Reply:
Make senses. I will remove.